speechbrain/lobes/models/huggingface_whisper.py

```diff
@@ -244,7 +245,7 @@ def _log_mel_spectrogram(self, audio):
         magnitudes = stft[..., :-1].abs() ** 2
 
         filters = self._mel_filters
-        mel_spec = filters @ magnitudes
+        mel_spec = filters.transpose(0, 1) @ magnitudes
```

Comment:
I don't understand why it is coming from the `transformers` library.... For me, it seems to be related to the `torch` library...

Reply:
Whats the torch and version are you using? I'll try the exact same version.
Can you try installing `torch 2.0` and `transformers 4.26` vs `transformers 4.32`. I can confirm that with I maintain the torch version throughout my expts and vary the transformer version, and the error vanishes as the per the table above.

Reply:
[log_transformer==4.30.2_script_changed.txt](https://github.com/speechbrain/speechbrain/files/11807596/log_transformer.4.30.2_script_changed.txt)
[log_transformer==4.30.2_script_notchanged.txt](https://github.com/speechbrain/speechbrain/files/11807597/log_transformer.4.30.2_script_notchanged.txt)
[log_transformer==4.26.0_script_notchanged.txt](https://github.com/speechbrain/speechbrain/files/11807598/log_transformer.4.26.0_script_notchanged.txt)

I attach the log files for your reference.

Reply:
in transformers >= 4.30, they change how they compute feature_extractor.mel_filters. As a result in transformers >= 4.30, the shape of feature_extractor.mel_filters is  (201,80) while in the prev version, it was (80,201).  It causes a problem in our _log_mel_spectrogram function ,we copy from openAI, when we calculate mel_spec = filters @ magnitudes. It expects the filter to be (80,201).

Reply:
Thanks Pooneh,I see. Do you know why they changed their 'feature_extractor.mel_filters' ?

Reply:
It is related to this PR on transformers:  https://github.com/huggingface/transformers/pull/21998

Basically what they do is replacing the hand-rolled STFT in the different models including whisper with the one from audio_util.   

In Transformers version <= 4.28:

The implementation in audio_utils for function "mel_filter_bank" which they use to get feature_extractor.mel_filters  is the same as  pytorchaudio function https://pytorch.org/audio/main/generated/torchaudio.functional.melscale_fbanks.html
This will return  (n_freqs, n_mels) --> (210, 80). In transformers-implementation for whisper 
in version <= 4.28 they use another function that is specific for whisper  --> "get_mel_filters "in /transformers/models/whisper/feature_extraction_whisper.py which return  (n_mels, n_freqs, ) --> (80, 210)

Reply:
It is related to this PR on transformers: 
 https://github.com/huggingface/transformers/pull/21998

Basically what they do is replacing the hand-rolled STFT in the different models including whisper with the one from audio_util.   

In Transformers version <= 4.28: 
In order to get  self.mel_filters , they use function that is specific for whisper  --> "get_mel_filters "in /transformers/models/whisper/feature_extraction_whisper.py which returns  (n_mels, n_freqs, ) --> (80, 210)
then, to calculate the log-Mel spectrogram, they call their own stft function in transformers/models/whisper/feature_extraction_whisper.py and finally call _np_extract_fbank_features which is basically the same as our function _log_mel_spectrogram in /speechbrain/lobes/models/huggingface_whisper.py which is the same as open_ai function.

In version >= 4.29:
In order to get the self.mel_filters they use  "mel_filter_bank" function  in  transformers/audio_utils.py.
It  is the same as  following pytorchaudio function:
https://pytorch.org/audio/main/generated/torchaudio.functional.melscale_fbanks.html
This will return  (n_freqs, n_mels) --> (210, 80).
Then they call spectrogram function in audio_utils.py to generate log_mel_spectrogram and they transpose  mel_filters.
spectrogram = np.maximum(mel_floor, np.dot(mel_filters.T, spectrogram)
This function is quite similar to melscale in Pytorchaudio
https://pytorch.org/audio/main/generated/torchaudio.transforms.MelScale.html

Final Points:
1) It seems they try to unify all audio functions across different models and use the same implementation as pytorchaudio and Librosa. These models are affected by this change:
 * CLAP
 * M-CTC-T
 * SpeechT5
 * TVLT
 * Whisper
 
2) Based on the implementation, I think the suggested fix by @sangeet2020  makes sense.
3) I am going to run all comon_voice recipes for the major release. Maybe it would be better to merge this change so I could test whisper recipes and make sure there won't be any bugs introduced by this change.

Reply:
@Adel-Moumen  and @sangeet2020  I am thinking about applying transpose only if the shape is not n_mels, n_freqs, So it could also work for an older version of transformers.